server/app/views/applicant/ApplicantUpsellView.java

```diff
@@ -0,0 +1,152 @@
+                    .withClasses(
+                        "flex",
+                        "flex-col",
+                        "gap-4",
+                        "justify-end",
+                        StyleUtils.responsiveMedium("flex-row")));
+
+    return Modal.builder()
+        .setModalId(Modal.randomModalId())
+        .setContent(modalContent)
+        .setTriggerButtonContent(
+            button(messages.at(triggerButtonMsg.getKeyName()))
+                .withClasses(ApplicantStyles.BUTTON_UPSELL_SECONDARY_ACTION))
+        .setModalTitle(modalTitle)
+        .setWidth(Width.HALF)
+        .build();
+  }
+
+  protected static ButtonTag createLoginButton(
+      String buttonId, Messages messages, String postLoginRedirectTo) {
+    return redirectButton(
+            buttonId,
+            messages.at(MessageKey.BUTTON_LOGIN.getKeyName()),
+            routes.LoginController.applicantLogin(Optional.of(postLoginRedirectTo)).url())
+        .withClasses(ApplicantStyles.BUTTON_UPSELL_SECONDARY_ACTION);
+  }
+
+  protected static ButtonTag createNewAccountButton(String buttonId, Messages messages) {
+    return redirectButton(
+            buttonId,
+            messages.at(MessageKey.BUTTON_CREATE_ACCOUNT.getKeyName()),
+            routes.LoginController.register().url())
+        .withClasses(ApplicantStyles.BUTTON_UPSELL_PRIMARY_ACTION);
+  }
+
+  protected static ButtonTag createApplyToProgramsButton(
+      String buttonId, String buttonText, Long applicantId) {
+    return redirectButton(
+            buttonId,
+            buttonText,
+            controllers.applicant.routes.ApplicantProgramsController.index(applicantId).url())
+        .withClasses(ApplicantStyles.BUTTON_UPSELL_PRIMARY_ACTION);
+  }
+
+  protected static DivTag createMainContent(
+      String title,
+      SectionTag confirmationSection,
+      Boolean shouldUpsell,
+      Messages messages,
+      String civicEntityFullName,
+      ImmutableList<DomContent> actionButtons) {
+    return div()
+        .withClasses(ApplicantStyles.PROGRAM_INFORMATION_BOX)
+        .with(
+            h1(title).withClasses("text-3xl", "text-black", "font-bold", "mb-4"),
+            confirmationSection,
+            createAccountManagementSection(
+                shouldUpsell, messages, civicEntityFullName, actionButtons));
+  }
+
+  protected static HtmlBundle createHtmlBundle(
+      ApplicantLayout layout,
+      String title,
+      Optional<ToastMessage> bannerMessage,
+      Modal loginPromptModal,
+      DivTag mainContent) {
+    HtmlBundle bundle = layout.getBundle().setTitle(title);
+    bannerMessage.ifPresent(bundle::addToastMessages);
+    bundle
+        .addMainStyles(ApplicantStyles.MAIN_PROGRAM_APPLICATION)
+        .addMainContent(mainContent)
+        .addModals(loginPromptModal);
+    return bundle;
+  }
+
+  /** Don't show "create an account" upsell box to TIs, or anyone with an email address already. */
```

Comment:
I think the comment is out of date as it references email address.

Reply:
Updated